src/server/main.ts

```diff
@@ -168,14 +168,21 @@ regUrlApi<MoonReq, MoonRes>({
   genReq: async (req) => `https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/timeline/${req.location}/${req.date}?unitGroup=metric&include=days&key=${moonApiKey}&contentType=json&elements=datetime,moonphase,sunrise,sunset,moonrise,moonset`,
   genRes: async (res) => res as MoonRes
 });
+regApi<FireballReq, FireballRes>({
```

Comment:
Use 'regUrlApi' instead, the generic 'regApi' function takes another function which extracts the request data type from the HTTP request.